pkg/kine/server/list.go

```diff
@@ -28,20 +26,17 @@ func (l *LimitedServer) list(ctx context.Context, r *etcdserverpb.RangeRequest)
 		return nil, fmt.Errorf("invalid range end length of 0")
 	}
 
-	prefix := string(append(r.RangeEnd[:len(r.RangeEnd)-1], r.RangeEnd[len(r.RangeEnd)-1]-1))
-	if !strings.HasSuffix(prefix, "/") {
-		prefix = prefix + "/"
-	}
-	start := string(bytes.TrimRight(r.Key, "\x00"))
+	start := r.Key
+	end := r.RangeEnd
 	revision := r.Revision
 	span.SetAttributes(
-		attribute.String("prefix", prefix),
-		attribute.String("start", start),
+		attribute.String("start", string(start)),
```

Comment:
Again, this allocates just for logging. Is there any alternative?

Reply:
The OpenTelemetry API doesn't support byte keys: https://pkg.go.dev/go.opentelemetry.io/otel/attribute#KeyValue

One alternative would be to check if OpenTelemetry is enabled and skip this completely otherwise: https://github.com/canonical/k8s-dqlite/blob/88a826e9e9c548cdc1c2ccb1e4279f74e3e63b7e/cmd/root.go#L32